src/algebra/order/complete_field.lean

```diff
@@ -0,0 +1,314 @@
+instance : conditionally_complete_linear_ordered_field ℝ :=
+{ ..real.linear_ordered_field, ..real.conditionally_complete_linear_order }
+
+namespace rat
+
+/-!
+### Rational cut map
+
+The idea is that a conditionally complete linear ordered field is fully characterized by its copy of
+the rationals. Hence we define `rat.cut_map β : α → set β` which sends `a : α` to the "rationals in
+`β`" that are less than `a`.
+-/
+
+section cut_map
+variables [linear_ordered_field α]
+
+section division_ring
+variables (β) [division_ring β] {a a₁ a₂ : α} {b : β} {q : ℚ}
+
+/-- The lower cut of rationals inside a linear ordered field that are less than a given element of
+another linear ordered field. -/
+def cut_map (a : α) : set β := (coe : ℚ → β) '' {t | ↑t < a}
+
+lemma cut_map_mono (h : a₁ ≤ a₂) : cut_map β a₁ ⊆ cut_map β a₂ := image_subset _ $ λ _, h.trans_lt'
+
+variables {β}
+
+@[simp] lemma mem_cut_map_iff : b ∈ cut_map β a ↔ ∃ q : ℚ, (q : α) < a ∧ (q : β) = b := iff.rfl
+
+@[simp] lemma coe_mem_cut_map_iff [char_zero β] : (q : β) ∈ cut_map β a ↔ (q : α) < a :=
+rat.cast_injective.mem_set_image
+
+lemma cut_map_self (a : α) : cut_map α a = Iio a ∩ range (coe : ℚ → α) :=
+begin
+  ext,
+  split,
+  { rintro ⟨q, h, rfl⟩,
+    exact ⟨h, q, rfl⟩ },
+  { rintro ⟨h, q, rfl⟩,
+    exact ⟨q, h, rfl⟩ }
+end
+
+end division_ring
+
+variables (β) [linear_ordered_field β] {a a₁ a₂ : α} {b : β} {q : ℚ}
+
+lemma cut_map_coe (q : ℚ) : cut_map β (q : α) = coe '' {r : ℚ | (r : β) < q} :=
+by simp_rw [cut_map, rat.cast_lt]
+
+variables [archimedean α]
+
+lemma cut_map_nonempty (a : α) : (cut_map β a).nonempty :=  nonempty.image _ $ exists_rat_lt a
+
+lemma cut_map_bdd_above (a : α) : bdd_above (cut_map β a) :=
+begin
+  obtain ⟨q, hq⟩ := exists_rat_gt a,
+  exact ⟨q, ball_image_iff.2 $ λ r hr, by exact_mod_cast (hq.trans' hr).le⟩,
+end
+
+lemma cut_map_add (a b : α) : cut_map β (a + b) = cut_map β a + cut_map β b :=
+begin
+  refine (image_subset_iff.2 $ λ q hq, _).antisymm _,
+  { rw [mem_set_of_eq, ←sub_lt_iff_lt_add] at hq,
+    obtain ⟨q₁, hq₁q, hq₁ab⟩ := exists_rat_btwn hq,
+    refine ⟨q₁, q - q₁, _, _, add_sub_cancel'_right _ _⟩; try {norm_cast}; rwa coe_mem_cut_map_iff,
+    exact_mod_cast sub_lt.mp hq₁q },
+  { rintro _ ⟨_, _, ⟨qa, ha, rfl⟩, ⟨qb, hb, rfl⟩, rfl⟩,
+    refine ⟨qa + qb, _, by norm_cast⟩,
+    rw [mem_set_of_eq, cast_add],
+    exact add_lt_add ha hb }
+end
+
+end cut_map
+
+/-!
+### Induced map
+
+`rat.cut_map` spits out a `set β`. To get something in `β`, we now take the supremum.
+-/
+
+section induced_map
+variables (α β γ) [linear_ordered_field α] [conditionally_complete_linear_ordered_field β]
+  [conditionally_complete_linear_ordered_field γ]
+
+/-- The induced order preserving function from a linear ordered field to a conditionally complete
+linear ordered field, defined by taking the Sup in the codomain of all the rationals less than the
+input. -/
+def induced_map (x : α) : β := Sup $ cut_map β x
```

Comment:
Is the full name `rat.induced_map`?
Since this map is unlikely to be used a lot, I would prefer to put this in `linear_ordered_field` or even `conditionally_complete_linear_ordered_field`.

Reply:
Yes it is. `induced_map` surely won't be used much but I can see its bundlings be, and they should share the same namespace. I quite like `rat`, but I don't like the `induced`. Unfortunately I have no other idea at the moment.